sql/catalyst/src/main/scala/org/apache/spark/sql/types/Decimal.scala

```diff
@@ -413,6 +413,18 @@ final class Decimal extends Ordered[Decimal] with Serializable {
             if (doubled > pow10diff || doubled == pow10diff && longVal % 2 != 0) {
               longVal += (if (droppedDigits < 0) -1L else 1L)
             }
+          case ROUND_UP =>
```

Comment:
Not sure about the changePrecision method for ROUND_UP, ROUND_DOWN and ROUND_HALF_DOWN. If anyone could help me on this that would be great.

Reply:
Also do you have any input on this please ? I need to integrate that in DecimalSuite.scala.

Reply:
In the current implementation, `changePrecision` should not take `ROUND_UP`, `ROUND_DOWN` and `ROUND_HALF_DOWN`. If we have such case, it's a bug so how about having an assertion before this `match` expression or simply falling to the default `_` branch?

Reply:
Hi @sarutak thanks for your review. Out of curiosity, may i know why it will be a bug if `changePrecision` takes `ROUND_UP`, `ROUND_DOWN` or `ROUND_HALF_DOWN` ?

So we could do something like this :
```
case ROUND_UP | ROUND_DOWN | ROUND_HALF_DOWN | _ =>
            throw QueryExecutionErrors.unsupportedRoundingMode(roundMode)
```

Reply:
> Out of curiosity, may i know why it will be a bug if changePrecision takes ROUND_UP, ROUND_DOWN or ROUND_HALF_DOWN ?

It's not expected that `ROUND_UP`, `ROUND_DOWN` and `ROUND_HALF_DOWN` are not expected to be passed to `changePrecision` and it's private.

Reply:
ok i see